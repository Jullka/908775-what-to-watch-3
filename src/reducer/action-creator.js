import DataActionCreator from '../reducer/data/data.js';
import UserActionCreator from '../reducer/user/user.js';

export default Object.assign({}, DataActionCreator, UserActionCreator);
