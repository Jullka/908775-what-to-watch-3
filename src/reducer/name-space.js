const NameSpace = {
  DATA: `DATA`,
  USER: `USER`
};

export {NameSpace};
