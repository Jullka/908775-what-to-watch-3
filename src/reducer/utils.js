import AppActionCreator from '../reducer/app/app.js';
import UserActionCreator from '../reducer/user/user.js';
import {AppState, AuthorizationStatus, ErrorCode} from '../components/const.js';

const handleError = (err) => (dispatch) => {
  if (err.response.status === ErrorCode.UNAUTHORIZED) {
    return dispatch(UserActionCreator.setAuthorizationStatus(AuthorizationStatus.NO_AUTH));
  }

  return dispatch(AppActionCreator.changeAppState(AppState.ERROR));
};

export {handleError};
