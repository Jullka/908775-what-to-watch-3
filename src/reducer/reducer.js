import {extend} from '../utils.js';
import {Movies, movieDetails} from '../mocks/movies.js';
import {getMoviesByGenre} from '../components/utils/get-movies-by-genre.js';
import {GameScreen} from '../components/const.js';

const ALL_GENRES = `All genres`;
const SHOWN_MOVIES_NUMBER = 8;

const initialState = {
  gameScreen: GameScreen.MAIN,
  history: [],
  movieDetails,
  selectedGenre: ALL_GENRES,
  movies: Movies,
  selectedMovie: null,
  filteredMovies: [...Movies],
  shownMoviesNumber: Movies.length > SHOWN_MOVIES_NUMBER ? SHOWN_MOVIES_NUMBER : Movies.length
};

const ActionType = {
  CHANGE_GAME_SCREEN: `CHANGE_GAME_SCREEN`,
  CHANGE_GENRE: `CHANGE_GENRE`,
  GET_MOVIES_BY_GENRE: `GET_MOVIES_BY_GENRE`,
  SELECT_MOVIE: `SELECT_MOVIE`,
  SHOW_MORE_MOVIES: `SHOW_MORE_MOVIES`,
  RESET_SHOW_MORE_MOVIES: `RESET_SHOW_MORE_MOVIES`,
  ROLLBACK: `ROLLBACK`
};

const ActionCreator = {
  changeGenre: (genre) => ({
    type: ActionType.CHANGE_GENRE,
    payload: genre
  }),
  getMoviesByGenre: () => ({
    type: ActionType.GET_MOVIES_BY_GENRE
  }),
  selectMovie: (movie) => ({
    type: ActionType.SELECT_MOVIE,
    payload: movie
  }),
  showMoreMovies: () => ({
    type: ActionType.SHOW_MORE_MOVIES,
    payload: SHOWN_MOVIES_NUMBER
  }),
  resetShowMoreMovies: () => ({
    type: ActionType.RESET_SHOW_MORE_MOVIES
  }),
  changeGameScreen: (gameScreen) => ({
    type: ActionType.CHANGE_GAME_SCREEN,
    payload: gameScreen
  }),
  rollback: () => ({
    type: ActionType.ROLLBACK
  })
};

const reducer = (state = initialState, action) => {
  switch (action.type) {
    case ActionType.CHANGE_GAME_SCREEN:
      return extend(state, {
        gameScreen: action.payload
      });
    case ActionType.CHANGE_GENRE:
      return extend(state, {
        selectedGenre: action.payload
      });

    case ActionType.GET_MOVIES_BY_GENRE:
      return extend(state, {
        filteredMovies: getMoviesByGenre(state.movies, state.selectedGenre),
      });

    case ActionType.SELECT_MOVIE:
      return extend(state, {
        selectedMovie: action.payload
      });
    case ActionType.SHOW_MORE_MOVIES:
      return extend(state, {
        shownMoviesNumber: state.shownMoviesNumber + action.payload
      });
    case ActionType.RESET_SHOW_MORE_MOVIES:
      return extend(state, {
        shownMoviesNumber: SHOWN_MOVIES_NUMBER
      });
    case ActionType.ROLLBACK:
      state.history.pop();
      state.history.pop();
      return reducer(state, state.history.pop());

    default:
      return reducer(state, {
        type: ActionType.CHANGE_GAME_SCREEN,
        payload: GameScreen.MAIN
      });
  }
};

export {reducer, ActionType, ActionCreator};
