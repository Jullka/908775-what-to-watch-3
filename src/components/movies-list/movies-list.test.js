import React from 'react';
import renderer from 'react-test-renderer';
import {reducer} from '../../reducer/reducer.js';
import {createStore} from 'redux';
import {Provider} from 'react-redux';
import MoviesList from '../movies-list/movies-list.jsx';

const Movies = [
  {
    id: `001`,
    title: `Fantastic Beasts: The Crimes of Grindelwald`,
    genre: `Thrillers`,
    releaseDate: 1991,
    poster: `img/johnny-english.jpg`,
    bigPoster: `img/moonrise-kingdom.jpg`,
    video: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
    ratingScore: 8.4,
    ratingLevel: `Very good`,
    ratingCount: 240,
    text: `Based on a true story about a small-time, self-possessed personal-injury attorney whose greed entangles him in a case that threatens to destroy him. The Woburn Case- which appears straightforward- instead evolves into a labyrinthine lawsuit of epic proportions where truth, if it can be found at all, resides not in the courtroom, but buried deep in a network of deceit and corruptions.`,
    director: `Steven Zaillian`,
    starring: `John Travolta, Robert Duvall, Stephen Fry and other`
  },
  {
    id: `002`,
    title: `Bohemian Rhapsody`,
    genre: `Thrillers`,
    releaseDate: 1991,
    poster: `img/johnny-english.jpg`,
    bigPoster: `img/moonrise-kingdom.jpg`,
    video: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
    ratingScore: 8.4,
    ratingLevel: `Very good`,
    ratingCount: 240,
    text: `Based on a true story about a small-time, self-possessed personal-injury attorney whose greed entangles him in a case that threatens to destroy him. The Woburn Case- which appears straightforward- instead evolves into a labyrinthine lawsuit of epic proportions where truth, if it can be found at all, resides not in the courtroom, but buried deep in a network of deceit and corruptions.`,
    director: `Steven Zaillian`,
    starring: `John Travolta, Robert Duvall, Stephen Fry and other`
  },
  {
    id: `003`,
    title: `Macbeth`,
    genre: `Thrillers`,
    releaseDate: 1991,
    poster: `img/johnny-english.jpg`,
    bigPoster: `img/moonrise-kingdom.jpg`,
    video: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
    ratingScore: 8.4,
    ratingLevel: `Very good`,
    ratingCount: 240,
    text: `Based on a true story about a small-time, self-possessed personal-injury attorney whose greed entangles him in a case that threatens to destroy him. The Woburn Case- which appears straightforward- instead evolves into a labyrinthine lawsuit of epic proportions where truth, if it can be found at all, resides not in the courtroom, but buried deep in a network of deceit and corruptions.`,
    director: `Steven Zaillian`,
    starring: `John Travolta, Robert Duvall, Stephen Fry and other`
  },
  {
    id: `004`,
    title: `Aviator`,
    genre: `Thrillers`,
    releaseDate: 1991,
    poster: `img/johnny-english.jpg`,
    bigPoster: `img/moonrise-kingdom.jpg`,
    video: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
    ratingScore: 8.4,
    ratingLevel: `Very good`,
    ratingCount: 240,
    text: `Based on a true story about a small-time, self-possessed personal-injury attorney whose greed entangles him in a case that threatens to destroy him. The Woburn Case- which appears straightforward- instead evolves into a labyrinthine lawsuit of epic proportions where truth, if it can be found at all, resides not in the courtroom, but buried deep in a network of deceit and corruptions.`,
    director: `Steven Zaillian`,
    starring: `John Travolta, Robert Duvall, Stephen Fry and other`
  },
  {
    id: `005`,
    title: `We need to talk about Kevin`,
    genre: `Thrillers`,
    releaseDate: 1991,
    poster: `img/johnny-english.jpg`,
    bigPoster: `img/moonrise-kingdom.jpg`,
    video: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
    ratingScore: 8.4,
    ratingLevel: `Very good`,
    ratingCount: 240,
    text: `Based on a true story about a small-time, self-possessed personal-injury attorney whose greed entangles him in a case that threatens to destroy him. The Woburn Case- which appears straightforward- instead evolves into a labyrinthine lawsuit of epic proportions where truth, if it can be found at all, resides not in the courtroom, but buried deep in a network of deceit and corruptions.`,
    director: `Steven Zaillian`,
    starring: `John Travolta, Robert Duvall, Stephen Fry and other`
  },
  {
    id: `006`,
    title: `What We Do in the Shadows`,
    genre: `Thrillers`,
    releaseDate: 1991,
    poster: `img/johnny-english.jpg`,
    bigPoster: `img/moonrise-kingdom.jpg`,
    video: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
    ratingScore: 8.4,
    ratingLevel: `Very good`,
    ratingCount: 240,
    text: `Based on a true story about a small-time, self-possessed personal-injury attorney whose greed entangles him in a case that threatens to destroy him. The Woburn Case- which appears straightforward- instead evolves into a labyrinthine lawsuit of epic proportions where truth, if it can be found at all, resides not in the courtroom, but buried deep in a network of deceit and corruptions.`,
    director: `Steven Zaillian`,
    starring: `John Travolta, Robert Duvall, Stephen Fry and other`
  },
  {
    id: `007`,
    title: `Revenant`,
    genre: `Thrillers`,
    releaseDate: 1991,
    poster: `img/johnny-english.jpg`,
    bigPoster: `img/moonrise-kingdom.jpg`,
    video: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
    ratingScore: 8.4,
    ratingLevel: `Very good`,
    ratingCount: 240,
    text: `Based on a true story about a small-time, self-possessed personal-injury attorney whose greed entangles him in a case that threatens to destroy him. The Woburn Case- which appears straightforward- instead evolves into a labyrinthine lawsuit of epic proportions where truth, if it can be found at all, resides not in the courtroom, but buried deep in a network of deceit and corruptions.`,
    director: `Steven Zaillian`,
    starring: `John Travolta, Robert Duvall, Stephen Fry and other`
  },
  {
    id: `008`,
    title: `Johnny English`,
    genre: `Thrillers`,
    releaseDate: 1991,
    poster: `img/johnny-english.jpg`,
    bigPoster: `img/moonrise-kingdom.jpg`,
    video: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
    ratingScore: 8.4,
    ratingLevel: `Very good`,
    ratingCount: 240,
    text: `Based on a true story about a small-time, self-possessed personal-injury attorney whose greed entangles him in a case that threatens to destroy him. The Woburn Case- which appears straightforward- instead evolves into a labyrinthine lawsuit of epic proportions where truth, if it can be found at all, resides not in the courtroom, but buried deep in a network of deceit and corruptions.`,
    director: `Steven Zaillian`,
    starring: `John Travolta, Robert Duvall, Stephen Fry and other`
  }
];

const handleMovieClick = () => {};

const store = createStore(reducer);

it(`MoviesList should render correctly`, () => {
  const tree = renderer
   .create(
       <Provider store={store}>
         <MoviesList
           movies={Movies}
           onMovieClick={handleMovieClick}
         />
       </Provider>,
       {
         createNodeMock: () => ({})
       }
   )
   .toJSON();

  expect(tree).toMatchSnapshot();
});
