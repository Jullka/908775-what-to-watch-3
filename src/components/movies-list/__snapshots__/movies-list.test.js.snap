// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MoviesList should render correctly 1`] = `
<div
  className="catalog__movies-list"
>
  <article
    className="small-movie-card catalog__movies-card"
    onClick={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    <div
      className="small-movie-card__image"
    >
      <video
        width="100%"
      />
    </div>
    <h3
      className="small-movie-card__title"
    >
      <a
        className="small-movie-card__link"
        href="movie-page.html"
        onClick={[Function]}
      >
        Fantastic Beasts: The Crimes of Grindelwald
      </a>
    </h3>
  </article>
  <article
    className="small-movie-card catalog__movies-card"
    onClick={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    <div
      className="small-movie-card__image"
    >
      <video
        width="100%"
      />
    </div>
    <h3
      className="small-movie-card__title"
    >
      <a
        className="small-movie-card__link"
        href="movie-page.html"
        onClick={[Function]}
      >
        Bohemian Rhapsody
      </a>
    </h3>
  </article>
  <article
    className="small-movie-card catalog__movies-card"
    onClick={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    <div
      className="small-movie-card__image"
    >
      <video
        width="100%"
      />
    </div>
    <h3
      className="small-movie-card__title"
    >
      <a
        className="small-movie-card__link"
        href="movie-page.html"
        onClick={[Function]}
      >
        Macbeth
      </a>
    </h3>
  </article>
  <article
    className="small-movie-card catalog__movies-card"
    onClick={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    <div
      className="small-movie-card__image"
    >
      <video
        width="100%"
      />
    </div>
    <h3
      className="small-movie-card__title"
    >
      <a
        className="small-movie-card__link"
        href="movie-page.html"
        onClick={[Function]}
      >
        Aviator
      </a>
    </h3>
  </article>
  <article
    className="small-movie-card catalog__movies-card"
    onClick={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    <div
      className="small-movie-card__image"
    >
      <video
        width="100%"
      />
    </div>
    <h3
      className="small-movie-card__title"
    >
      <a
        className="small-movie-card__link"
        href="movie-page.html"
        onClick={[Function]}
      >
        We need to talk about Kevin
      </a>
    </h3>
  </article>
  <article
    className="small-movie-card catalog__movies-card"
    onClick={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    <div
      className="small-movie-card__image"
    >
      <video
        width="100%"
      />
    </div>
    <h3
      className="small-movie-card__title"
    >
      <a
        className="small-movie-card__link"
        href="movie-page.html"
        onClick={[Function]}
      >
        What We Do in the Shadows
      </a>
    </h3>
  </article>
  <article
    className="small-movie-card catalog__movies-card"
    onClick={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    <div
      className="small-movie-card__image"
    >
      <video
        width="100%"
      />
    </div>
    <h3
      className="small-movie-card__title"
    >
      <a
        className="small-movie-card__link"
        href="movie-page.html"
        onClick={[Function]}
      >
        Revenant
      </a>
    </h3>
  </article>
  <article
    className="small-movie-card catalog__movies-card"
    onClick={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    <div
      className="small-movie-card__image"
    >
      <video
        width="100%"
      />
    </div>
    <h3
      className="small-movie-card__title"
    >
      <a
        className="small-movie-card__link"
        href="movie-page.html"
        onClick={[Function]}
      >
        Johnny English
      </a>
    </h3>
  </article>
</div>
`;
